packages/hardhat/contracts/FundingVaultFactory.sol

```diff
@@ -99,23 +113,54 @@ contract FundingVaultFactory{
         _withdrawlAddress,
         _developerFeeAddress, 
         _developerFeePercentage, 
-        _projectURL
+        _projectURL,
+        _projectTitle,
+        _projectDescription
         );
 
-        bool tokenTransferSuccess = participationToken.transferFrom(msg.sender, address(fundingVault), _participationTokenAmount);
-        if (!tokenTransferSuccess){
-            revert FundingVault__TokenTransferFailed();
-        }
+        
+        transferParticipationTokens(msg.sender, address(fundingVault), _participationTokenAmount);
 
-        s_fundingVaults[fundingVaultId] = address(fundingVault);
+        Vault storage vault = vaults[fundingVaultId];
+        vault.vaultAddress = address(fundingVault);
+        vault.title = _projectTitle;
+        vault.description = _projectDescription;
+        vault.deadline = _blockLimit;
+        
         emit FundingVaultDeployed(address(fundingVault));     
         return address(fundingVault);
     }
 
+     /**
+     * @notice Get list of all funding vaults
+     * @dev to access the list of all the available funding vaults on the platform 
+     */ 
+    function getVaults(uint256 start, uint256 end) external view returns(Vault[] memory)
+    {
+        if(end > s_fundingVaultIdCounter || start > end || start == 0)
+        {
+            revert FundingVaultFactory__InvalidIndex();
+        }
+        Vault[] memory allVaults = new Vault[](end - start + 1);
+
+        for(uint i = start; i <= end;i++)
+        {
+            allVaults[i - start] = vaults[i];
+        }
+        return allVaults;
+    }
+
+    function transferParticipationTokens(address from, address to, uint256 amount) private{
+        bool tokenTransferSuccess = participationToken.transferFrom(from, to, amount);
+        if (!tokenTransferSuccess){
+            revert FundingVault__TokenTransferFailed();
+        }
+    }
+
 
     // Getters //
     function getFundingVault(uint256 _fundingVaultId) external view returns (address) {
-        return s_fundingVaults[_fundingVaultId];
+        return vaults[_fundingVaultId].vaultAddress;
     }
 
     function getTotalNumberOfFundingVaults() external view returns (uint256) {
```

Comment:
Remove this function and make the variable public instead.

Reply:
Fixed